include/pm/device_runtime.h

```diff
@@ -8,141 +9,118 @@
  *
  * @funcprops \pre_kernel_ok
  *
- * @param dev Pointer to device structure of the specific device driver
- * the caller is interested in.
+ * @param dev Device instance.
+ *
+ * @retval 0 If the device runtime PM is disabled successfully.
+ * @retval -ENOSYS If the functionality is not available.
+ * @retval -errno Other negative errno, result of resuming the device.
  */
-void pm_device_disable(const struct device *dev);
+int pm_device_runtime_disable(const struct device *dev);
 
 /**
- * @brief Call device resume asynchronously based on usage count
+ * @brief Resume a device based on usage count.
  *
- * Called by a device driver to mark the device as being used.
- * This API will asynchronously bring the device to resume state
- * if it not already in active state.
+ * This function will resume the device if the device is suspended (usage count
+ * equal to 0). In case of a resume failure, usage count and device state will
+ * be left unchanged. In all other cases, usage count will be incremented.
  *
- * @funcprops \isr_ok, \pre_kernel_ok
+ * If the device is still being suspended as a result of calling
+ * pm_device_runtime_put_async(), this function will wait for the operation to
+ * finish to then resume the device.
  *
- * @param dev Pointer to device structure of the specific device driver
- * the caller is interested in.
- * @retval 0 If successfully queued the Async request. If queued,
- * the caller need to wait on the poll event linked to device
- * pm signal mechanism to know the completion of resume operation.
- * @retval Errno Negative errno code if failure.
- */
-int pm_device_get_async(const struct device *dev);
-
-/**
- * @brief Call device resume synchronously based on usage count
+ * @funcprops \pre_kernel_ok
  *
- * Called by a device driver to mark the device as being used. It
- * will bring up or resume the device if it is in suspended state
- * based on the device usage count. This call is blocked until the
- * device PM state is changed to resume.
+ * @param dev Device instance.
  *
- * @param dev Pointer to device structure of the specific device driver
- * the caller is interested in.
- * @retval 0 If successful.
- * @retval Errno Negative errno code if failure.
+ * @retval 0 If the device has been resumed successfully.
+ * @retval -ENOSTUP If runtime PM is not enabled for the device.
+ * @retval -ENOSYS If the functionality is not available.
+ * @retval -errno Other negative errno, result of the PM action callback.
  */
-int pm_device_get(const struct device *dev);
+int pm_device_runtime_get(const struct device *dev);
 
 /**
- * @brief Call device suspend asynchronously based on usage count
+ * @brief Suspend a device based on usage count.
+ *
+ * This function will suspend the device if the device is no longer required
+ * (usage count equal to 0). In case of suspend failure, usage count and device
+ * state will be left unchanged. In all other cases, usage count will be
+ * decremented (up to 0).
```

Comment:
"down to 0" ;-)
Otherwise, this is fine to me

Reply:
done